react/src/Admin/components/CreateUserModal.tsx

```diff
@@ -19,55 +20,87 @@ export interface CreateUserModalProps {
     onSuccess: (state: NewUser) => void;
 }
 
+const useStyles = makeStyles(theme => ({
+    submitting: {
+        cursor: "wait",
+    },
+}));
+
+function ErrorText(props: { id: string; errorCode: number }) {
+    switch (props.errorCode) {
+        case 0:
+            return <></>;
+        case 400:
+            return (
+                <DialogContentText id={props.id} color="secondary">
+                    Username or email too long, bad email, or other invalid input.
+                </DialogContentText>
+            );
+        case 404:
+            return (
+                <DialogContentText id={props.id} color="secondary">
+                    Some requested permission groups do not exist.
+                </DialogContentText>
+            );
+        case 422:
+            return (
+                <DialogContentText id={props.id} color="secondary">
+                    Username or email already in use.
```

Comment:
I think it is really useful for users to know which field is already in use, so it would be nice if we can differentiate between email and username

Reply:
LOL revert part of 1038fc8

Reply:
I went a slightly different route and added something that's like what we'll want #219 to do for all endpoints.